gui.py

```diff
@@ -0,0 +1,189 @@
+"""
+This module contains implementation of graphical user interface, using PySide6
+
+"""
+from functools import lru_cache
+from typing import Generator
+import math
+
+from PySide6 import QtWidgets, QtGui, QtCore
+
+from model import GameMap, GameState
+from presenter import Presenter
+from config import ui as ucf
+from coordinates import Coordinates
+
+
+class Window(QtWidgets.QMainWindow):
+    """
+    Main window of user interface
+    """
+    def __init__(self, login_data, parent=None):
+        super().__init__(parent)
+        self.presenter_thread = Presenter(login_data)
+        self.presenter_thread.start()
+        self.screen = QtWidgets.QApplication.screenAt(self.pos())
+        self.setGeometry(0, 0, self.screen.size().width()//2, self.screen.size().height()//1.1)
```

Comment:
Some magic numbers. It might be not clear why 2 and 1.1. It is better to use named constants.